pkmntypes.py

```diff
@@ -22,3 +22,30 @@ def __init__(self, **kwargs):
 
 class Pokemon():
 	pass
+
+class BattleContext():
+	"""This is a class that describes the point of view of a given Pokemon on the battlefield. It provides
+	enough information for a user to make an informed decision about what turn to make next. 
+
+	:param battle: The state of the battlefield
+	:type battle: dict
+	:param pokemon: The pokemon that this context belongs to
+	:type pokemon: dict
+	:param team: The team ID of the `Pokemon`
+	:type team: dict
+	:param targets: All pokemon on the battlefield
+	:type targets: dict
+	:param allies: Ally targets in relation to the `Pokemon`
+	:type allies: dict
+	:param opponents: Enemy targets in relation to the `Pokemon`
+	:type opponents: dict
+	"""
+	
+	def __init__(self, **kwargs):
+
+		self.battle = kwargs['Battle']
```

Comment:
...because they should be specified down here.

eg. `self.battle: dict = kwargs['Battle']`

Reply:
OHHHHHHHHHH